### PR TITLE
[ch85] As a user, I can see homepage structure explicitly

### PR DESCRIPTION
**Pull request by LilBank**
https://app.clubhouse.io/cobda/story/85/as-a-user-i-can-see-homepage-structure-explicitly

## Why?

This is an update for Home Screen to see each sections with grid implementation.

To describe the reason for creating this **PR**.

## Changes

- Add layout for home page.

### Desktop
Screen Shot 2564-01-25 at 23 52 51

### Mobile
Screen Shot 2564-01-25 at 23 53 04

### CODE REVIEW

File: src/components/HomeSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      </section>
+      <section className="home-section">
+        <header className="home-section__header">
+          <h2 className="home-section__title">Sample 4</h2>
+          <p className="home-section__description">This is a sample description</p>
+        </header>
+      </section>
+    </>
+  );
+}
```
Comment (kimvcp) [inline src/components/HomeSection/index.tsx]: Please add a newline (Resolve when fixed)
```suggestion
}

```

File: src/screens/Home/index.tsx [hunk 2]
```diff
@@ -1,16 +1,21 @@
 import React from 'react';
+import { useEffect } from 'react';
```
Comment (kimvcp) [inline src/screens/Home/index.tsx]: They are from the same module. We should group them up
```suggestion
import React, { useEffect } from 'react';
```

File: src/scss/components/_home-section.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    &__header {}
+
+    &__title {
+        margin: 0 0 rem(24px) 0;
+    }
+
+    &__description {
+        margin: 0;
+    }
+}
```
Comment (kimvcp) [inline src/scss/components/_home-section.scss]: Newline needed (Resolve when fixed)
```suggestion
}

```

File: src/scss/layouts/_layout-home.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        @include respond-above('lg') {
+            padding: 5% 10%;
+        }
+    }
+
+    & > footer {
+        grid-area: footer;
+    }
+}
```
Comment (kimvcp) [inline src/scss/layouts/_layout-home.scss]: Newline needed (Resolve when fixed)
```suggestion
}

```

**Review by kimvcp — CHANGES_REQUESTED** [review]

File: src/scss/components/_home-section.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    // TODO: For future implementation
+    &__header {}
+
+    &__title {
+        margin: 0 0 rem(24px) 0;
+    }
+
+    &__description {
+        margin: 0;
+    }
```
Comment (kimvcp) [inline src/scss/components/_home-section.scss]: Please use two indents for the tab size.

```suggestion
  width: 100%;
  height: rem(200px);
  background-color: aquamarine;
  padding: rem(16px);
  box-sizing: border-box;

  @include respond-above('md') {
    padding: rem(24px);
    height: rem(320px);
  }

  @include respond-above('lg') {
    padding: rem(32px);
    height: rem(400px);
  }

  &--dark {
    background-color: blue;
  }

  &--light {
    background-color: blueviolet;
  }

  // TODO: For future implementation
  &__header {
  }

  &__title {
    margin: 0 0 rem(24px) 0;
  }

  &__description {
    margin: 0;
  }
```

File: src/scss/layouts/_layout-home.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        @include respond-above('lg') {
+            padding: 5% 10%;
+        }
+    }
+
+    & > footer {
+        grid-area: footer;
+    }
```
Comment (kimvcp) [inline src/scss/layouts/_layout-home.scss]: This one also needs two indents for tab size

```suggestion
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: rem(32px) 1fr rem(48px);
  grid-template-areas:
    'header'
    'main'
    'footer';

  height: 100vh;

  @include respond-above('md') {
    grid-template-rows: rem(48px) 1fr rem(72px);
  }

  @include respond-above('lg') {
    grid-template-rows: rem(72px) 1fr rem(100px);
  }

  & > header {
    grid-area: header;
  }

  & > main {
    grid-area: main;

    padding: rem(24px) rem(16px);

    @include respond-above('md') {
      padding: 4.5%;
    }

    @include respond-above('lg') {
      padding: 5% 10%;
    }
  }

  & > footer {
    grid-area: footer;
  }
```

File: src/scss/layouts/__layouts-dir.scss [hunk 7]
```diff
@@ -0,0 +1 @@
+@import './layout-home';
```
Comment (kimvcp) [inline src/scss/layouts/__layouts-dir.scss]: Newline needed (Resolve when fixed)
```suggestion
@import './layout-home';

```

**Review by Kittinske15 — APPROVED** [review]

File: src/scss/layouts/_layout-home.scss [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        grid-template-rows: rem(72px) 1fr rem(100px);
+      }
+
+    & > header {
+        grid-area: header;
+    }
+
+    & > main {
+        grid-area: main;
+
```
Comment (Raksani) [inline src/scss/layouts/_layout-home.scss:19]: Does it need the extra space here?

**Review by Raksani — CHANGES_REQUESTED** [review]

File: src/scss/layouts/_layout-home.scss (hunk 8 above)
Comment (LilBank) [reply src/scss/layouts/_layout-home.scss:19]: Sure, just for grid.

File: src/scss/layouts/_layout-home.scss [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    padding: rem(24px) rem(16px);
+
+    @include respond-above('md') {
+        padding: 4.5% 6%;
+    }
+
+    @include respond-above('lg') {
+        padding: 5% 10%;
+    }
+  }
```
Comment (kimvcp) [inline src/scss/layouts/_layout-home.scss]: This one also needs 2 indents.

```suggestion
    @include respond-above('md') {
      padding: 4.5% 6%;
    }

    @include respond-above('lg') {
      padding: 5% 10%;
    }
  }
```

**Review by Kittinske15 — APPROVED** [review]

**Review by kimvcp — APPROVED** [review]

**Pull request merged**